modules/extension/xsd/xsd-wfs/src/main/java/org/geotools/wfs/v2_0/bindings/EnvelopePropertyTypeBinding.java

```diff
@@ -19,20 +19,41 @@
 import java.util.ArrayList;
 import java.util.List;
 import javax.xml.namespace.QName;
+import net.opengis.wfs20.EnvelopePropertyType;
+import net.opengis.wfs20.Wfs20Factory;
+import net.opengis.wfs20.Wfs20Package;
+import org.eclipse.emf.ecore.EAttribute;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EcoreFactory;
+import org.eclipse.emf.ecore.util.FeatureMap;
+import org.eclipse.emf.ecore.util.FeatureMap.Entry;
+import org.eclipse.emf.ecore.util.FeatureMapUtil;
 import org.eclipse.xsd.XSDElementDeclaration;
 import org.geotools.geometry.jts.ReferencedEnvelope;
 import org.geotools.gml3.v3_2.GML;
 import org.geotools.wfs.v2_0.WFS;
 import org.geotools.xsd.AbstractComplexBinding;
+import org.geotools.xsd.ElementInstance;
+import org.geotools.xsd.Node;
 
 public class EnvelopePropertyTypeBinding extends AbstractComplexBinding {
 
+    private Class<ReferencedEnvelope> referencedEnvelopeClass = ReferencedEnvelope.class;
```

Comment:
All of these... they look like they should be constants, no?

Reply:
Hi Andrea,

could be constants, but in that case on class loading of EnvelopePropertyTypeBinding the EMF Factories would also be bootstrapped. In my experience that sometimes causes trouble, maybe in testing scenarios, etc.
On the other hand I was not sure about the lifespan of the EnvelopePropertyTypeBinding. If it is created often, statics might make sense...
As you like. Let me know what you prefer please.

Best regards,
Andreas

Reply:
If memory serves me right, bindings are instantiated anew every time we parse or encode something. Each parse/encode gets its own dedicated Pico Container context, bindings are registered in there, every time. I'm not sure if they are just registered as classes, and then instantiated every time they are needed, or not. If it was the former, that would mean they get instantiated multiple times per parse/encode operation, too.

Reply:
Do you prefer static fields then?

Reply:
Personally I do, yes.